tests/helpers/generateFilterType.spec.ts

```diff
@@ -0,0 +1,119 @@
+import expect from "expect";
+
+import {
+  IntrospectionInputObjectType,
+  graphql,
+  getIntrospectionQuery,
+  IntrospectionSchema,
+  TypeKind,
+  IntrospectionNamedTypeRef
+} from "graphql";
+import {
+  Field,
+  Resolver,
+  Query,
+  buildSchema,
+  ObjectType,
+  Int,
+  Arg
+} from "type-graphql";
+import { generateFilterType } from "../../lib/helpers";
+import { Filter } from "../../lib/decorators";
+
+describe("generateFilterType", () => {
+  let schemaIntrospection: IntrospectionSchema;
+
+  before(async () => {
+    @ObjectType("SomeName")
+    class FilterableType {
+      @Field(type => Int)
+      @Filter(["lt", "gt"], type => Int)
+      amount: number;
+
+      @Field(type => String, { name: "purpose" })
+      @Filter(["eq", "like"])
+      getPurpose(): string {
+        return "some purpose";
+      }
+    }
+
+    @Resolver()
+    class SampleResolver {
+      @Query(() => FilterableType)
+      filterableQuery(
+        @Arg("filter", generateFilterType(FilterableType), { nullable: true })
```

Comment:
I am now less happy about the name of this function 🙁 because in fact it generates the return type function for the filter, not just the filter type, but `generateReturnTypeFunc` is a bit more complex of a name so I think I'll leave it like this.

Unless you have a better idea?

Reply:
Let's improve that name once we find a better one 😄